### PR TITLE
♻️ Refactor testing button

### PR DESCRIPTION
**Pull request by MarioBatalha**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206135681624099

### CODE REVIEW

File: src/lib/components/button/stories/index.stories.tsx [hunk 1]
```diff
@@ -276,14 +911,41 @@ export const ButtonWithAction: Story = {
...
+    await step('Hover button test', async () => {
+      await userEvent.hover(canvas.getByRole('button'));
+    });
 
-    //click on button
-    await step('click on button', () => {
-      button.click();
+    await step('Unhover button test', async () => {
+      await userEvent.unhover(canvas.getByRole('button'));
     });
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:933]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 2]
```diff
@@ -252,16 +856,47 @@ export const Icon: Story = {
...
+      await userEvent.hover(canvas.getByRole('button'));
+    });
+
+    await step('Unhover button test', async () => {
+      await userEvent.unhover(canvas.getByRole('button'));
+    });
+
+    await step('Button args test', async () => {
+      expect(args.variant).toBe('filled');
+    });
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:893]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 3]
```diff
@@ -243,6 +808,45 @@ export const TonalWithCombined: Story = {
...
+      await userEvent.dblClick(canvas.getByRole('button'));
+    });
+
+    await step('Hover button test', async () => {
+      await userEvent.hover(canvas.getByRole('button'));
+    });
+
+    await step('Unhover button test', async () => {
+      await userEvent.unhover(canvas.getByRole('button'));
+    });
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:834]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 4]
```diff
@@ -233,6 +759,45 @@ export const TonalWithSuffix: Story = {
...
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:785]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 5]
```diff
@@ -224,6 +711,45 @@ export const Tonal: Story = {
...
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:737]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 6]
```diff
@@ -216,6 +664,45 @@ export const SmallTextWithCombined: Story = {
...
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:690]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 7]
```diff
@@ -206,6 +615,45 @@ export const TextWithCombined: Story = {
...
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:641]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 8]
```diff
@@ -196,6 +566,45 @@ export const TextWithSuffix: Story = {
...
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:592]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 9]
```diff
@@ -187,6 +518,45 @@ export const TextWithPrefix: Story = {
...
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:544]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 10]
```diff
@@ -178,6 +470,45 @@ export const Text: Story = {
...
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:496]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 11]
```diff
@@ -170,6 +423,45 @@ export const OutlineWithCombined: Story = {
...
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:449]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 12]
```diff
@@ -160,6 +374,45 @@ export const OutlineWithSuffix: Story = {
...
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:400]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 13]
```diff
@@ -151,6 +326,45 @@ export const OutlineWithPrefix: Story = {
...
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:352]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 14]
```diff
@@ -142,6 +278,45 @@ export const Outline: Story = {
...
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:304]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 15]
```diff
@@ -125,14 +229,46 @@ export const FilledWithCombined: Story = {
...
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:246]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 16]
```diff
@@ -95,10 +163,42 @@ export const FilledWithSuffix: Story = {
...
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:180]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 17]
```diff
@@ -69,10 +100,42 @@ export const FilledWithPrefix: Story = {
...
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:117]: Find one way check if this event occurr on the elements

File: src/lib/components/button/stories/index.stories.tsx [hunk 18]
```diff
@@ -34,19 +34,45 @@ export const Filled: Story = {
...
```
Comment (KipandaJr) [inline src/lib/components/button/stories/index.stories.tsx:61]: Find one way check if this event occurr on the elements

**Review by KipandaJr — CHANGES_REQUESTED** [review]

File: src/lib/components/button/stories/index.stories.tsx (hunk 18 above)
Comment (MarioBatalha) [reply src/lib/components/button/stories/index.stories.tsx:61]: Screenshot 2024-06-19 at 17 54 43

Comment (MarioBatalha) [reply src/lib/components/button/stories/index.stories.tsx:61]: @KipandaJr when you validate this, please resolve the conversation

**Review by atilasousa — APPROVED** [review]

**Pull request merged**